src/code/z_message.c

```diff
@@ -64,46 +322,246 @@
+        XREG(74) = 0x6A;
+        XREG(75) = 0;
+        XREG(77) = 0x58;
+        XREG(76) = 0x44;
+    } else {
+        XREG(74) = 0x42;
+        XREG(75) = 0x1E;
+        XREG(77) = 0x3C;
+        XREG(76) = 0x1C;
+        msgCtx->unk11F1A[0] = (msgCtx->unk11F1A[1] = (msgCtx->unk11F1A[2] = 0));
+        Interface_ChangeAlpha(1U);
+    }
+}
+
+void func_80151BB4(GlobalContext* globalCtx, u8 uParm2) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    u8 temp = uParm2;
+
+    if (CHECK_QUEST_ITEM(QUEST_BOMBERS_NOTEBOOK)) {
+        if (((gSaveContext.weekEventReg[D_801C6B28[uParm2] >> 8]) & (u8)D_801C6B28[uParm2]) == 0) {
+            msgCtx->unk120B2[msgCtx->unk120B1] = temp;
+            msgCtx->unk120B1++;
+        }
+    } else if (uParm2 >= 20) {
+        if (((gSaveContext.weekEventReg[D_801C6B28[uParm2] >> 8]) & (u8)D_801C6B28[uParm2]) == 0) {
+            msgCtx->unk120B2[msgCtx->unk120B1] = temp;
+            msgCtx->unk120B1++;
+        }
+    }
+}
+
+u32 func_80151C9C(GlobalContext* globalCtx) {
+    MessageContext* msgCtx;
+    u8 flag;
+
+    msgCtx = &globalCtx->msgCtx;
+    while (1) {
+        if (msgCtx->unk120B1 == 0) {
+            return 0;
+        }
+        msgCtx->unk120B1--;
+
+        if (((gSaveContext.weekEventReg[D_801C6B28[msgCtx->unk120B2[msgCtx->unk120B1]] >> 8]) &
+             (u8)D_801C6B28[msgCtx->unk120B2[msgCtx->unk120B1]]) == 0) {
+            flag = gSaveContext.weekEventReg[D_801C6B28[msgCtx->unk120B2[msgCtx->unk120B1]] >> 8];
+            gSaveContext.weekEventReg[D_801C6B28[msgCtx->unk120B2[msgCtx->unk120B1]] >> 8] =
+                flag | (u8)D_801C6B28[msgCtx->unk120B2[msgCtx->unk120B1]];
+            if ((D_801C6AB8[msgCtx->unk120B2[msgCtx->unk120B1]] != 0) && (CHECK_QUEST_ITEM(QUEST_BOMBERS_NOTEBOOK))) {
+                func_80151938(globalCtx, D_801C6AB8[msgCtx->unk120B2[msgCtx->unk120B1]]);
+                play_sound(NA_SE_SY_SCHEDULE_WRITE);
+                return 1;
+            }
+        }
+    }
+}
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80151DA4.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80152434.s")
+void func_80152434(GlobalContext* globalCtx, u16 arg2) {
+    globalCtx->msgCtx.unk12046 = 0;
+    func_80151DA4(globalCtx, arg2);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80152464.s")
+void func_80152464(GlobalContext* globalCtx, u16 arg1) {
+    globalCtx->msgCtx.unk12046 = 1;
+    func_80151DA4(globalCtx, arg1);
+}
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_message/Message_GetState.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_8015268C.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80152C64.s")
+void func_80152C64(View* view) {
+    Viewport viewport;
+    // clang-format off
+    viewport.bottomY = 0xF0;viewport.rightX = 0x140;
+    // clang-format on
+    viewport.topY = 0;
+    viewport.leftX = 0;
+
+    View_SetViewport(view, (Viewport*)&viewport);
```

Comment:
This looks like the `SET_FULLSCREEN_VIEWPORT` macro. Would that one match here?

Reply:
Thanks, didn't know that macro. done in 26329f8e